city_scrapers/spiders/fortx_Tarrant_County_Commissioners_Court.py

```diff
@@ -0,0 +1,95 @@
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+from dateutil.parser import parse as dateparse
+
+
+class FortxTarrantCountyCommissionersCourtSpider(CityScrapersSpider):
+    name = "fortx_Tarrant_County_Commissioners_Court"
+    agency = "Tarrant County Commissioners Court"
+    timezone = "America/Chicago"
+
+    """
+    The scraper gathers meeting information from two separate endpoints
+    specified in start_urls. One endpoint provides archived meetings,
+    and the other provides current and upcoming meetings. This approach
+    was chosen to keep the scraper simple and avoid overcomplicating it,
+    especially since the CSS of the source webpages was quite messy.
+    """
+    start_urls = [
+        "https://tarrant-agendamanagement-public.techsharetx.gov/publicportal/api/meetings/readArchived",  # noqa
+        "https://tarrant-agendamanagement-public.techsharetx.gov/publicportal/api/meetings/readCurrentAndUpcoming",  # noqa
+    ]
+    agenda_base_url = "https://prod-agendamanagement-publicportal.azurewebsites.us/HtmlAgenda/"  # noqa
+    committee_id = "fe6aa5cc-7448-4194-ac6e-08dc95f79ccc"
+
+    source_url = "https://www.tarrantcountytx.gov/en/commissioners-court/commissioners-court-agenda-videos.html"  # noqa
+
+    location = {
+        "address": "100 East Weatherford Street, 5th Floor, Fort Worth, Texas 76196",  # noqa
+        "name": "Tarrant County Administration Building (check the agenda for room location)",  # noqa
+    }
+
+    def start_requests(self):
+        payload = {"committeeId": self.committee_id}
+        for url in self.start_urls:
+            yield scrapy.Request(
+                url=url,
+                method="POST",
+                body=json.dumps(payload),
+                headers={"Content-Type": "application/json"},
+                callback=self.parse,
+            )
+
+    def parse(self, response):
+        data = json.loads(response.text)
+        meetings = data.get("data", [])
+        for item in meetings:
+            meeting = Meeting(
+                title=item.get("description", "Commissioners Court"),
+                description="",
+                classification=COMMISSION,
+                start=self._parse_datetime(item.get("meetingStartDateTime")),
+                end=self._parse_datetime(item.get("meetingEndDateTime")),
+                all_day=False,
+                time_notes="",
+                location=self.location,
+                links=self._parse_links(item),
+                source=self.source_url,
+            )
+
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    def _parse_datetime(self, datetime_str):
+        """
+        Parse the datetime string into a datetime object.
+        """
+        if datetime_str:
+            return dateparse(datetime_str)
+        return None
+
+    def _parse_links(self, item):
```

Comment:
I noticed that in addition to the agenda ID and video ID, the fetched data also includes the minutes ID. I believe we can construct the agenda URL and minutes URL using `agendaAttachmentId` and `minutesAttachmentId` with the following base URL: `https://tarrant-agendamanagement-public.techsharetx.gov/publicportal/api/meetingattachments/download?id={id}` (e.g. https://tarrant-agendamanagement-public.techsharetx.gov/publicportal/api/meetingattachments/download?id=c170623f-9a45-453c-017b-08dcfda5b8b4). What are your thoughts on this approach?

Reply:
Thanks for the feedback! I didn’t realize there was a single URL structure that could be used for both the agenda and minutes attachments. That’s a great insight. I'll refactor the `_parse_links` logic.